pkg/ami/ssm_resolver.go

```diff
@@ -57,6 +59,26 @@ func MakeSSMParameterName(version, instanceType, imageFamily string) (string, er
 		return fmt.Sprintf("/aws/service/ami-windows-latest/Windows_Server-2019-English-Core-EKS_Optimized-%s/%s", version, fieldName), nil
 	case api.NodeImageFamilyWindowsServer2019FullContainer:
 		return fmt.Sprintf("/aws/service/ami-windows-latest/Windows_Server-2019-English-Full-EKS_Optimized-%s/%s", version, fieldName), nil
+	case api.NodeImageFamilyWindowsServer2022CoreContainer:
+		const minVersion = api.Version1_23
+		supportsWindows2022, err := utils.IsMinVersion(minVersion, version)
+		if err != nil {
+			return "", err
+		}
+		if !supportsWindows2022 {
+			return "", errors.Errorf("Windows Server 2022 Core requires EKS version %s and above", minVersion)
+		}
+		return fmt.Sprintf("/aws/service/ami-windows-latest/Windows_Server-2022-English-Core-EKS_Optimized-%s/%s", version, fieldName), nil
+	case api.NodeImageFamilyWindowsServer2022FullContainer:
+		const minVersion = api.Version1_23
+		supportsWindows2022, err := utils.IsMinVersion(minVersion, version)
+		if err != nil {
+			return "", err
+		}
+		if !supportsWindows2022 {
+			return "", errors.Errorf("Windows Server 2022 Full requires EKS version %s and above", minVersion)
+		}
+		return fmt.Sprintf("/aws/service/ami-windows-latest/Windows_Server-2022-English-Full-EKS_Optimized-%s/%s", version, fieldName), nil
```

Comment:
The logic for `Windows Server 2022 Core` and `Windows Server 2022 Full` only differs in the SSM parameter name for the AMI family. 

`/aws/service/ami-windows-latest/Windows_Server-2022-English-<ami-type>-EKS_Optimized-%s/%s`

This can be factored out into a function so the case statements here just need to pass `Core` or `Full`.

Reply:
Hi Chetan,
I have refactored code according to your suggestion.
I have extracted out function to decide Windows AMI Type Core/Full based on imageFamily.